src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -686,181 +674,105 @@ bool ModifierManager::parse_scripts(DefinitionManager const& definition_manager)
+	return _expect_modifier_effect_with_fallback(registry, modifier_value, key_value_invalid_callback);
+}
+NodeTools::key_value_callback_t ModifierManager::_expect_modifier_effect_with_fallback(
+	CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque> const& registry,
+	ModifierValue& modifier_value,
+	NodeTools::key_value_callback_t const& fallback
+) const {
+	return [this, &registry, &modifier_value, fallback = FWD(fallback)](auto const& key, auto const& value) {
+		if (dryad::node_has_kind<ast::ListValue>(value) && complex_modifiers.contains(key)) {
+			const key_value_callback_t callback = [this, &modifier_value, &key](auto const& inner_key, auto const& value) -> bool{
+				return _add_flattened_modifier_cb(modifier_value, key, inner_key, value);
+			};
+			return expect_dictionary(callback)(value);
 		}
-
-		return default_callback(key, value);
+		
+		ModifierEffect const* effect = registry.get_item_by_identifier(key);
+		if(effect == nullptr) { return fallback(key, value); }
+		return _add_modifier_cb(modifier_value, effect, value);
 	};
 }
+NodeTools::key_value_callback_t ModifierManager::_expect_shared_tech_country_modifier_effect(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect(shared_tech_country_modifier_effects, modifier_value);
+}
+NodeTools::key_value_callback_t ModifierManager::expect_leader_modifier(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect(leader_modifier_effects, modifier_value);
+}
+NodeTools::key_value_callback_t ModifierManager::expect_technology_modifier(ModifierValue& modifier_value) const {
+	return [this, &modifier_value](auto const& key, auto const& value) {
+		if (strings_equal_case_insensitive(key, "rebel_org_gain")) { // because of course there's a special one
+			std::string_view faction_identifier;
+			ast::NodeCPtr value_node = nullptr;
+
+			bool ret = expect_dictionary_keys(
+				"faction", ONE_EXACTLY, expect_identifier(assign_variable_callback(faction_identifier)),
+				"value", ONE_EXACTLY, assign_variable_callback(value_node)
+			)(value);
 
-node_callback_t ModifierManager::expect_validated_modifier_value_and_default(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, key_value_callback_t default_callback,
-	ModifierEffectValidator auto effect_validator
-) const {
-	return [this, modifier_callback, type, default_callback, effect_validator](ast::NodeCPtr root) -> bool {
-		ModifierValue modifier;
-
-		bool ret = expect_dictionary_reserve_length(
-			modifier.values,
-			_modifier_effect_callback(modifier, type, default_callback, effect_validator)
-		)(root);
+			ret &= _add_flattened_modifier_cb(modifier_value, key, faction_identifier, value_node);
 
-		ret &= modifier_callback(std::move(modifier));
+			return ret;
+		}
 
-		return ret;
+		return _expect_modifier_effect_with_fallback(technology_modifier_effects, modifier_value,
+		_expect_shared_tech_country_modifier_effect(modifier_value)
+		)(key, value);
```

Comment:
```suggestion
		return _expect_modifier_effect_with_fallback(
			technology_modifier_effects, modifier_value,
			_expect_shared_tech_country_modifier_effect(modifier_value)
		)(key, value);
```
Formatting, and the same for the functions below